### PR TITLE
LibreELEC support

### PR DESCRIPTION
**Pull request by neox387**
This will add LibreELEC support,

However official releases do not work, this is also broken for me on your build for openelec, and getting a datetime error from beautifulsoup4 at line 118 in builds.py:

`return dict((unicode(iter_contents.next().string), tag['datetime'])
                    for tag in iter_contents`

### CODE REVIEW

**Comment by LS80** [comment]
Thanks, I will review this.

File: resources/lib/builds.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ def __repr__(self):
 
 class BuildLinkExtractor(BaseExtractor):
     """Base class for extracting build links from a URL"""
-    BUILD_RE = (".*OpenELEC.*-{arch}-(?:\d+\.\d+-|)[a-zA-Z]+-(\d+)"
+    BUILD_RE = (".*{release}.*-{arch}-(?:\d+\.\d+-|)[a-zA-Z]+-(\d+)"
```
Comment (LS80) [inline resources/lib/builds.py]: Probably `dist` or `distribution` is a better name than `release`.

File: resources/lib/builds.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def __iter__(self):
         if self.CSS_CLASS is not None:
             args.append(self.CSS_CLASS)
 
-        self.build_re = re.compile(self.BUILD_RE.format(arch=arch))
+        self.build_re = re.compile(self.BUILD_RE.format(release=openelec.OS_RELEASE['NAME'],arch=arch))
```
Comment (LS80) [inline resources/lib/builds.py]: How about a `dist` function in the `openelec` module which just returns `OS_RELEASE['NAME']`. That would read better I think.

Comment (LS80) [inline resources/lib/builds.py]: Missing space after the comma

File: resources/lib/builds.py [hunk 3]
```diff
@@ -262,15 +262,18 @@ def _create_link(self, link):
 
 
 class OfficialReleaseLinkExtractor(ReleaseLinkExtractor):
-    BASE_URL = "http://releases.openelec.tv"
+    if openelec.OS_RELEASE['NAME'] == "OpenELEC":
+        BASE_URL = "http://releases.openelec.tv"
+    elif openelec.OS_RELEASE['NAME'] == "LibreELEC":
+        BASE_URL = "http://releases.libreelec.tv"
```
Comment (LS80) [inline resources/lib/builds.py]: Could be reduced to one line
```
BASE_URL = "http://releases.{dist}.tv".format(dist=openelec.dist().lower())
```

File: resources/lib/builds.py [hunk 4]
```diff
@@ -412,8 +421,12 @@ def __repr__(self):
 class MilhouseBuildsURL(BuildsURL):
     def __init__(self, subdir="master"):
         self.subdir = subdir
+        if openelec.OS_RELEASE['NAME'] == "OpenELEC":
+            url = "http://milhouse.openelec.tv/builds/"
+        elif openelec.OS_RELEASE['NAME'] == "LibreELEC":
+            url = "http://milhouse.libreelec.tv/builds/"
```
Comment (LS80) [inline resources/lib/builds.py]: Could be reduced to one line similar to the comment above.

File: resources/lib/builds.py [hunk 5]
```diff
@@ -428,7 +441,7 @@ def get_installed_build():
     """Return the currently installed build object."""
     DEVEL_RE = "devel-(\d+)-r\d+-g([a-z0-9]+)"
 
-    if openelec.OS_RELEASE['NAME'] == "OpenELEC":
+    if openelec.OS_RELEASE['NAME'] == "OpenELEC" or openelec.OS_RELEASE['NAME'] == "LibreELEC":
```
Comment (LS80) [inline resources/lib/builds.py]: Could be 
```
if openelec.dist() in ("OpenELEC", "LibreELEC"):
```

File: resources/lib/openelec.py [hunk 6]
```diff
@@ -6,15 +6,14 @@
...
-
 try:
     ARCH = OS_RELEASE['OPENELEC_ARCH']
 except KeyError:
     # Enables testing on non OpenELEC machines
     ARCH = 'RPi.arm'
 
 UPDATE_DIR = os.path.join(os.path.expanduser('~'), '.update')
-if OS_RELEASE['NAME'] != "OpenELEC":
+if OS_RELEASE['NAME'] != "OpenELEC" and OS_RELEASE['NAME'] != "LibreELEC":
```
Comment (LS80) [inline resources/lib/openelec.py]: Could be
```
if openelec.dist() not in ("OpenELEC", "LibreELEC"):
```

**Review by LS80 — CHANGES_REQUESTED** [review]
Thanks for the PR. I've requested a few changes. Once we get this merged I'll probably fork it to make a LibreELEC specific version with the correct artwork.

**Comment by LS80** [comment]
I'll have a look at the problem with Official Releases.

**Comment by neox387** [comment]
It was just a quick dirty fix months ago :) & recently change to v18 forum post.

All tough LibreElec now has a build-in updater but I have not used it yet, I think it doesn't give a changelog not sure ..

**Pull request closed without merging**